chia/cmds/wallet_funcs.py

```diff
@@ -170,6 +173,167 @@ async def add_token(args: dict, wallet_client: WalletRpcClient, fingerprint: int
+            print("REQUESTING:")
+            for name, info in printable_dict.items():
+                amount, unit, multiplier = info
+                if multiplier > 0:
+                    print(f"  - {amount} {name} ({int(Decimal(amount) * unit)} mojos)")
+
+            confirmation = input("Confirm (y/n): ")
+            if confirmation not in ["y", "yes"]:
+                print("Not creating offer...")
+            else:
+                offer, trade_record = await wallet_client.create_offer_for_ids(offer_dict, fee=fee)
+                if offer is not None:
+                    with open(pathlib.Path(filepath), "w") as file:
+                        file.write(bytes(offer).hex())
+                        file.close()
+                    print(f"Created offer with ID {trade_record.trade_id}")
+                    print(f"Use chia wallet get_offers --id {trade_record.trade_id} -f {fingerprint} to view status")
+                else:
+                    print("Error creating offer")
+
+
+def timestamp_to_time(timestamp):
+    return datetime.fromtimestamp(timestamp).strftime("%Y-%m-%d %H:%M:%S")
+
+
+async def print_offer_summary(wallet_client: WalletRpcClient, sum_dict: dict):
+    for asset_id, amount in sum_dict.items():
+        if asset_id == "xch":
+            wid: str = "1"
+            name: str = "XCH"
+            unit: int = units["chia"]
+        else:
+            result = await wallet_client.cat_asset_id_to_name(bytes.fromhex(asset_id))
+            wid = "Unknown"
+            name = asset_id
+            unit = units["colouredcoin"]
+            if result is not None:
+                wid = str(result[0])
+                name = result[1]
+        print(f"    - {name} (Wallet ID: {wid}): {Decimal(int(amount)) / unit}")
+
+
+async def print_trade_record(record, wallet_client: WalletRpcClient, summaries: bool = False) -> None:
+    print()
+    print(f"Record with id: {record.trade_id}")
+    print("---------------")
+    print(f"Created at: {timestamp_to_time(record.created_at_time)}")
+    print(f"Confirmed at: {record.confirmed_at_index}")
+    print(f"Accepted at: {timestamp_to_time(record.accepted_at_time) if record.accepted_at_time else 'N/A'}")
+    print(f"Status: {TradeStatus(record.status).name}")
+    if summaries:
+        print("Summary:")
+        offer = Offer.from_bytes(record.offer)
+        offered, requested = offer.summary()
+        print("  OFFERED:")
+        await print_offer_summary(wallet_client, offered)
+        print("  REQUESTED:")
+        await print_offer_summary(wallet_client, requested)
+        print("Pending Balances:")
+        await print_offer_summary(wallet_client, offer.get_pending_amounts())
+        print(f"Fees: {Decimal(offer.bundle.fees()) / units['chia']}")
+    print("---------------")
+
+
+async def get_offers(args: dict, wallet_client: WalletRpcClient, fingerprint: int) -> None:
+    id, filepath, all, summaries = tuple(list(args.values())[0:4])
```

Comment:
I'd probably just make `id, filepath, all, summaries` function params to avoid this complex conversion/extraction.

Reply:
You can't do them as function arguments, they have to come in via that dictionary.  It's potentially more readable to do it in four lines, I just wasn't sure.